src_lockfree/michael_scott_queue.ml

```diff
@@ -75,7 +72,7 @@ let rec fix_tail tail new_tail =
     && not (Atomic.compare_and_set tail old_tail new_tail)
   then fix_tail tail new_tail
 
-let push { tail; _ } value =
+let push_exn { tail; _ } value =
```

Comment:
Hmm... The convention has been to use the `_exn` suffix on operations that raise an expected exception in some normal cases (such as when the queue is empty or full), but in this case no such exception is being used.  Why should we use the `_exn` suffix in this case?

Reply:
Good catch ! I made the change on `saturn_benchmarks` to match the generic `Queue` signature.  I should not have move it here !